react/features/conference/components/native/NavigationBar.js

```diff
@@ -55,26 +76,66 @@ class NavigationBar extends Component<Props> {
                 <SafeAreaView
                     pointerEvents = 'box-none'
                     style = { styles.navBarSafeView }>
-                    <View
-                        pointerEvents = 'box-none'
-                        style = { styles.navBarWrapper }>
-                        <PictureInPictureButton
-                            styles = { styles.navBarButton } />
-                        <View
-                            pointerEvents = 'box-none'
-                            style = { styles.roomNameWrapper }>
-                            <Text
-                                numberOfLines = { 1 }
-                                style = { styles.roomName }>
-                                { this.props._meetingName }
-                            </Text>
-                        </View>
-                    </View>
+                    { _navBarVisible && this._renderNavigationBar() }
+                    { _notificationsVisible
+                            && this._renderNotificationsContainer() }
                 </SafeAreaView>
             </View>
         );
     }
 
+    /**
+     * Renders a container for notifications to be displayed by the
+     * base/notifications feature.
+     *
+     * @private
+     * @returns {React$Element}
+     */
+    _renderNotificationsContainer() {
+        const notificationsStyle = {};
+
+        // In the landscape mode (wide) there's problem with notifications being
+        // shadowed by the filmstrip rendered on the right. This makes the "x"
+        // button not clickable. In order to avoid that a margin of the
+        // filmstrip's size is added to the right.
+        //
+        // Pawel: after many attempts I failed to make notifications adjust to
+        // their contents width because of column and rows being used in the
+        // flex layout. The only option that seemed to limit the notification's
+        // size was explicit 'width' value which is not better than the margin
+        // added here.
+        if (this.props._filmstripVisible && !isNarrowAspectRatio(this)) {
+            notificationsStyle.marginRight = FILMSTRIP_SIZE;
+        }
+
+        return <NotificationsContainer style = { notificationsStyle } />;
```

Comment:
I really prefer the components to be self-containing, meaning the component itself could also decide if we're in narrow or wide view, or there is filmstrip or not. If we do that, we don't need to pass the style prop from the outside

Reply:
But that will couple potentially unrelated features which are filmstrip and notifications. This will force the notification feature to know about the outer world. I was trying to not make the notifications component be aware of where is it displayed. It is the parent component that knows about all other external factors.

Reply:
I can accept this reasoning, but you also have to see that we're nor using the view type and fimstrip info acquired from the NavigationBar to where (based on your logic) it doesn't belong either. So you may be right, but we're already down that path :)

Anyhow, I leave it up to you to decide if you wanna change it or not. This may change in the future anyhow.